packaging/debian/storagenode.config

```diff
@@ -144,41 +146,65 @@ case "$STATE" in
             else
                   go_back
             fi
-      ;;
+	    ;;
       "$IP_ADDRESS")
-            db_input high storagenode/external_addr || true
-            if db_go; then
-                  db_get storagenode/external_addr
-                  ip_port="$RET"
-                  ip="${ip_port%:*}"
-                  port="${ip_port##*:}"
-
-                  if ! valid_ip $ip ;  then
-                        error_ask_continue storagenode/external_addr/not_ip_port_continue
-                  elif ! valid_port $port ;  then
-                        error_ask_continue storagenode/external_addr/not_ip_port_continue
-                  fi
-                  next_state
-            else
-                  go_back
-            fi
-      esac
+          db_input high storagenode/external_addr || true
+          if db_go; then
+              db_get storagenode/external_addr
+              ip_port="$RET"
+              ip="${ip_port%:*}"
+              port="${ip_port##*:}"
+
+              if ! valid_ip $ip ;  then
+                  error_ask_continue storagenode/external_addr/not_ip_port_continue
+              elif ! valid_port $port ;  then
+                  error_ask_continue storagenode/external_addr/not_ip_port_continue
+              fi
+              next_state
+          else
+              go_back
+          fi
+	  ;;
+      "$STORAGEDIR")
+          db_input high storagenode/storagedir || true
+          if db_go; then
+              db_get storagenode/storagedir
+              storagedir="$RET"
+
+	      if [ ! -d "$storagedir" ]; then
+		  error_ask_continue storagenode/storagedir/does_not_exist
+              fi
+              next_state
+          else
+              go_back
+          fi
+	  ;;
+      "$DISKSPACE")
+          db_input high storagenode/diskspace || true
+          if db_go; then
+              db_get storagenode/diskspace
+              diskspace="$RET"
+	      if [[ ! "$diskspace" =~ ^[0-9]+[GTgt]{1} ]]; then
+		  error_ask_continue storagenode/diskspace/wrong_format
+	      else
+		  input_diskspace_bytes=`numfmt --from=iec $diskspace`
+		  min_diskspace_bytes=`numfmt --from=iec 550G` # minimum diskspace with overhead
+
+		  if [ "$input_diskspace_bytes" -lt "$min_diskspace_bytes" ]; then
+		      error_ask_continue storagenode/diskspace/below_minimum
+		  fi
+		  if [ "$ERR_STATE" != "$ERR_BACK" ]; then
```

Comment:
I see that this is used across this file but shouldn't be better to use `if [ "$ERR_STATE" == "$ERR_CONTINUE" ]`